### DISCUSSION TITLE
Some suggestions for better usablity

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [GriefPrevention/GriefPrevention#506] [Ideas]
I have some suggestions to make the plugin feel more streamlines 
- option for automatic claim resizing when players build 
- option for disabling gold blocks around claim
- make it so there is a /claim command to start claim without needing a chest
- make it so there is a /claimshow command to show the players claim in particles
- maybe a icon or text for showing whos claim your in on the hud

automatic claim resizing would cost money and claim points and would notify the player if they don't have enough and are placing outside of their claim.

**Comment by RoboMWM** [comment]
FYI, there already is a `/claim` command. It does require holding the claim tool for now though.

**Comment by ShaneBeee** [comment]
Im going to give my feedback on each of these, as someone who loves this plugin both as a server operator and a player
1) Im 50/50 on this. I do see how this could be handy, but in my opinion, GP is about controlling your claim, learning how to take care of your land, etc. Its your job as the player to make sure to claim your land, and protect it.  
If an auto-claim system was in place, I could see this eating up more server power as the plugin would have to check resize claims potentially each and every time a blocks is placed. Not to mention if a player is standing on the edge of their claim and placing something meaningless (like a grass/dirt block they accidentally broke) this would force the claim to expand, when they might not wish it to do that. Also this could be bad as claims might get to be large when they aren't meant to be.

2) I don't really have an opinion here. I mean, they are there for a reason, turning them off then you just won't know where your claim is. 

3) As Robo said, this is already a thing. 

4) I disagree with the command aspect of this, the lovely thing is the player can do so much with the size/visuals of a claim without needing a command. I personally hate using commands, so using the stick in shovel for me is golden. I played on a server that used some weird WG land claim system, where it showed particles around the boundary of the claim, I just found it super laggy as the amount of particles was high. Obviously this could be controlled better. I do like the idea of it since often times its hard to see the corners of the claim if your land isn't relatively flat (ie: underground or custom generated land [EpicWorldGenerator as an example])

5) This can basically be done via GriefPrevention Flags using the enter/exit message flag. You can have it send a message to the player when they enter the claim. I did this on my server and it would say a message like "You entered ShaneBee's claim - You are safe from PvP"

**Comment by Shadowblitz16** [comment]
@ShaneBeee all of these would be optional
3. I don't like using items because they are used for different things and it requires you to have them with you constantly. for me having a command would be easier and more convenient
4. again it would be a optional thing you don't have to hide blocks and give them the permission to use the show command.
5. I would have to have a example

**Comment by TomLewis** [comment]
> make it so there is a /claimshow command to show the players claim in particles

A better way to do this would be to support WorldGuard CUI, I have asked for this before but I cant remember what the outcome was.
Being able to run a command for it to show in WorldGuard CUI would be MUCH better than using particles and no client performance drops, and more accurate.

**Comment by Shadowblitz16** [comment]
chunk protection shows a message on screen when you enter and leave your protected chunk I think that would be a better then ugly gold blocks.

as for the item, I think if a custom item with a custom texture was added I would be fine with it.
I just don't like having to use a existing item which for me is confusing.

**Comment by RoboMWM** [comment]
> chunk protection shows a message on screen when you enter and leave your protected chunk I think that would be a better then ugly gold blocks.

There are addons that provide this functionality already though, like GriefPrevention flags.

> as for the item, I think if a custom item with a custom texture was added I would be fine with it. I just don't like having to use a existing item which for me is confusing.

I've heard of others adding lore and such to their claim tool. Since it only does a material check, you could use any variant of it that you want, similar to how you'd define a custom item that's still seen as the original item by the server.

**Comment by Shadowblitz16** [comment]
> > chunk protection shows a message on screen when you enter and leave your protected chunk I think that would be a better then ugly gold blocks.
> 
> There are addons that provide this functionality already though, like GriefPrevention flags.
> 
Does it get rid of the gold blocks though?
> > as for the item, I think if a custom item with a custom texture was added I would be fine with it. I just don't like having to use a existing item which for me is confusing.
> 
> I've heard of others adding lore and such to their claim tool. Since it only does a material check, you could use any variant of it that you want, similar to how you'd define a custom item that's still seen as the original item by the server.
>
how would I do this? I don't think I can do this without a plugin

**Comment by TomLewis** [comment]
- Allowing WECUI support for showing the claim
